### PR TITLE
Turn off informer for statefulset unit tests

### LINKED ISSUE #106615: StatefulSet PVC tests failing
### Which jobs are failing?

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106568/pull-kubernetes-unit/1462136832413143040

### Which tests are failing?

StatefulSet control tests

### Since when has it been failing?

The delete PVC unit tests started flaking when https://github.com/kubernetes/kubernetes/pull/99728 was merged

### Testgrid link

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106568/pull-kubernetes-unit/1462136832413143040

### Reason for failure (if possible)

#99728 exposed a sharp edge in UpdateStatefulSet, where the set was assumed to be unmodified, but actually creating a new revision for it changes it.

### Anything else we need to know?

_No response_

### Relevant SIG(s)

/sig apps

/assign @mattcary

### PR DESCRIPTION
**Pull request by mattcary**
/kind failing-test

Fixes #106615

The situation is explained in the new comment in stateful_set_control_test.go.

```release-note
None
```

### CODE REVIEW

File: pkg/controller/statefulset/stateful_set_utils.go [hunk 1]
```diff
@@ -233,6 +233,9 @@ func updateClaimOwnerRefForSetAndPod(claim *v1.PersistentVolumeClaim, set *apps.
 			needsUpdate = removeOwnerRef(claim, pod) || needsUpdate
 		}
 	case policy.WhenScaled == delete && policy.WhenDeleted == delete:
+		// An object won't be garbage collected until all of its owners have gone away. So
+		// if a pod is scaling down, the set owner ref must be removed for the claim to be
+		// deleted with the pod.
```
Comment (liggitt) [inline pkg/controller/statefulset/stateful_set_utils.go:238]: is this comment relevant to this PR?

**Comment by mattcary** [comment]
ping?

**Comment by liggitt** [comment]
would like an apps reviewer to ack

**Comment by mattcary** [comment]
> would like an apps reviewer to ack

I've pinged the sig-apps slack 🤞🏾🤞🏾🤞🏾

File: pkg/controller/statefulset/stateful_set_control_test.go [hunk 2]
```diff
@@ -53,23 +53,33 @@ import (
...
-		informerFactory.Apps().V1().StatefulSets().Informer().HasSynced,
-		informerFactory.Core().V1().Pods().Informer().HasSynced,
-		informerFactory.Apps().V1().ControllerRevisions().Informer().HasSynced,
-	)
-	return om, ssu, ssc, stop
+	// The informer is not started. The tests here manipulate the local cache (indexers) directly, and there is no waiting
+	// for client state to sync. In fact, because the client is not updated during tests, informer updates will break tests
+	// by unexpectedly deleting objects.
+	//
+	// It would probably be better to rewrite all these tests manipulate the client an explicitly sync to ensure consistent
```
Comment (janetkuo) [inline pkg/controller/statefulset/stateful_set_control_test.go:68]: Make it a TODO?

**Review by janetkuo — APPROVED** [review]

File: pkg/controller/statefulset/stateful_set_control_test.go (hunk 2 above)
Comment (mattcary) [reply pkg/controller/statefulset/stateful_set_control_test.go:68]: Done

**Pull request merged**